tests/units/test_check_users.py

```diff
@@ -0,0 +1,243 @@
+            data = self.list_users_response3
+        else:
+            data = 'No data'
+
+        return ContextualStringIO(data)
+
+    def side_effect_function3(self, *args, **kwargs):
+        if 'users_to_delete.txt' in args[0]:
+            data = self.list_user_to_delete
+        elif 'list_role_assignments_response1.json' in args[0]:
+            data = self.list_role_assignment2
+        elif 'list_users_response3.json' in args[0]:
+            data = self.list_users_response3
+        else:
+            data = 'No data'
+
+        return ContextualStringIO(data)
+
+    @patch('fiwareskuld.utils.osclients.session', mock_session)
+    def test_check_user_init(self):
+        from fiwareskuld.check_users import CheckUsers
+
+        expectedUserSet = set(['fake1', 'fake2', 'fake3'])
+        expectedUserBasicSet = set([u'fake1', u'fake2'])
+
+        with patch('__builtin__.open') as mocked_open:
+            mocked_open.side_effect = self.side_effect_function
+
+            check = CheckUsers()
+
+            self.assertSetEqual(expectedUserSet, check.ids,
+                                "Unexpected content in users_to_delete file")
+
+            self.assertSetEqual(expectedUserBasicSet, check.users_basic,
+                                "The expected users with Basic role is not obtained")
+
+    @patch('fiwareskuld.utils.osclients.session', mock_session)
+    def test_report_not_basic_users(self):
+        from fiwareskuld.check_users import CheckUsers
+
+        expectedUserSet = set(['fake3'])
+        expectedUserTypeSet = set(['community'])
+
+        with patch('__builtin__.open') as mocked_open:
+            mocked_open.side_effect = self.side_effect_function
+
+            check = CheckUsers()
+
+            no_basic_users, no_basic_userstype = check.report_not_basic_users()
+
+            self.assertSetEqual(expectedUserSet, no_basic_users,
+                                "The expected set of no basic users is wrong")
+
+            self.assertSetEqual(expectedUserTypeSet, no_basic_userstype,
+                                "The expected set of type of no basic users is wrong")
+
+    @patch('fiwareskuld.utils.osclients.session', mock_session)
+    def test_report_invalid_users_type(self):
+        from fiwareskuld.check_users import CheckUsers
+
+        expectedUserSet = set(['fake3'])
+        expectedUserTypeSet = set(['unkown'])
+
+        with patch('__builtin__.open') as mocked_open:
+            mocked_open.side_effect = self.side_effect_function2
+
+            check = CheckUsers()
+
+            no_basic_users, no_basic_userstype = check.report_not_basic_users()
+
+            self.assertSetEqual(expectedUserSet, no_basic_users,
+                                "The expected set of no basic users is wrong")
+
+            self.assertSetEqual(expectedUserTypeSet, no_basic_userstype,
+                                "The expected set of type of no basic users is wrong")
```

Comment:
you should add assert validating mock is called: self.assertTrue(mock_open.called)